bin/prefix-vendor-namespace.php

```diff
@@ -57,39 +60,51 @@
 			continue 2;
 		}
 
+		$namespace_change = 0;
+		$uses_change      = 0;
+
 		$contents =	preg_replace(
 			"#^(\s*)(namespace)\s*({$quoted}[\\\\|;])#m",
 			"\$1\$2 {$new_namespace}\\\\\$3",
-			$contents
+			$contents,
+			-1,
+			$namespace_change
 		);
 
 		$contents =	preg_replace(
 			"#^(\s*)(use)\s*({$quoted}\\\\)#m",
 			"\$1\$2 {$new_namespace}\\\\\$3",
-			$contents
+			$contents,
+			-1,
+			$uses_change
 		);
 
 		if ( ! empty( $direct_replacements[ $path ] ) ) {
 			foreach ( $direct_replacements[ $path ] as $replace ) {
-				$replace  = preg_quote( $replace, '#' );
-				$contents =	preg_replace(
+				$direct_change = 0;
+				$replace       = preg_quote( $replace, '#' );
+				$contents      = preg_replace(
 					"#({$replace})#m",
 					"{$new_namespace}\\\\\$1",
-					$contents
+					$contents,
+					-1,
+					$direct_change
 				);
+				if ( 0 < $direct_change ) {
+					$uses_change += $direct_change;
+				}
 			}
 		}
+		if ( 0 === $namespace_change && 0 < $uses_change ) {
+			$file_notices[] = $file;
+		}
 
 		file_put_contents( $file, $contents );
 	}
 
 	// Update the namespace in the composer.json files.
-	$composer_files = array_filter(
-		explode(
-			"\n",
-			`find "{$vendor_dir}/{$path}" -iname 'composer.json'`
-		)
-	);
+
+	$composer_files = get_dir_contents($vendor_dir . DIRECTORY_SEPARATOR . $path, '/composer.json/');
```

Comment:
Is the `DIRECTORY_SEPARATOR` explicitly needed on Windows? There are other locations in this file where we are explicitly using `/` as the directory separator.

Also since we want to explicitly match the full file name shouldn't the pattern be: `/^composer.json$/`

Reply:
Regarding `DIRECTORY_SEPARATOR`, for reading file paths it isn't necessary (only when matching file paths - more on that later). So I reverted these constants to just `'/'` ([nice WordPress conversation on it here](https://core.trac.wordpress.org/ticket/44654)).

Regarding the explicit match of `composer.json`, you're right that the regex needs to be tweaked to ensure `sample-composer.json` isn't matched. However the `^` caused the match to fail, because there was a full path before it. Prefixing it with `/` only works on OS' that use that as the path separator, so using a `preg_quote` around `DIRECTORY_SEPARATOR` creates a way to ensure the file name is just `composer.json` while matching it on any OS.